osu.Game/Online/API/Requests/GetBeatmapSetsResponse.cs

```diff
@@ -65,7 +65,7 @@ public BeatmapSetInfo ToBeatmapSet(RulesetStore rulesets)
                     Ranked = ranked,
                     LastUpdated = lastUpdated,
                 },
-                Beatmaps = beatmaps.Select(b => b.ToBeatmap(rulesets)).ToList(),
+                Beatmaps = beatmaps?.Select(b => b.ToBeatmap(rulesets)).ToList(),

```

Comment:
null check should be unnecessary

Reply:
![image](https://user-images.githubusercontent.com/12570757/33244610-1d6d1846-d2fb-11e7-81e8-8821466f38b8.png)
I used `GetBeatmapSetsResponse` because the json response matches it, only dates, an author id and _beatmaps_ are missing

Reply:
did you reply to the wrong comment here? my issue was an unnecessary null check.

Reply:
`beatmaps` can be null because I'm using `GetBeatmapsSetsResponse` in place where the json doesn't have a beatmaps array

Reply:
could you provide an example?

Reply:
the request for the most played beatmaps of a user returns a beatmapset json object which overlaps with `GetBeatmapSetsResponse`. That is why I use it for the response class. However it doesn't have a beatmaps key which is why `beatmaps` is null in this case. I didn't think creating a seperate class just for the beatmapset json object would be a good idea because adding this null check seemed easier.
https://pastebin.com/892hHJwz

Reply:
but you split it out now, right? so why is it still null checking?

Reply:
https://github.com/ppy/osu/pull/1532/files#diff-5be61c40bdd29d04ef22a93898b0a6afR37
no...?

Reply:
Okay, misunderstood because the naming is all over the place. You can't use a class called `xxxResponse` inside another class called `xxxResponse`. At *very least*, `GetBeatmapSetsResponse` should be called `APIResponseBeatmap` or something